floodsub.go

```diff
@@ -319,6 +365,32 @@ func msgID(pmsg *pb.Message) string {
 	return string(pmsg.GetFrom()) + string(pmsg.GetSeqno())
 }
 
+// validate is called in a goroutine and calls the validate functions of all subs with msg as parameter.
+func (p *PubSub) validate(subs []*Subscription, msg *Message) bool {
+	for _, sub := range subs {
+		ctx, cancel := context.WithTimeout(p.ctx, sub.validateTimeout)
+		defer cancel()
+
+		result := make(chan bool)
+		go func(sub *Subscription) {
+			result <- sub.validate == nil || sub.validate(ctx, msg)
```

Comment:
maybe should return an error from the validator? Could be useful to log it

Reply:
My thinking was that the validator is user-supplied and they can do all the logging they want themselves. No need to make the API complex.